src/adam7.rs

```diff
@@ -285,6 +382,434 @@ fn test_adam7() {
+
+    assert_eq!(img, expected_img);
+
+    // Fifth pass
+    splat_expand_pass(img, stride, &[0b00011011u8], &it.next().unwrap(), bits_pp);
+    splat_expand_pass(img, stride, &[0b11100100u8], &it.next().unwrap(), bits_pp);
+    copy_nbytes_mtimes(&mut expected_img[4..], &[0b00000101, 0b10101111], 2, 2);
+    copy_nbytes_mtimes(&mut expected_img[12..], &[0b11111010, 0b01010000], 2, 2);
+
+    assert_eq!(img, expected_img);
+
+    // Sixth pass
+    splat_expand_pass(img, stride, &[0b00011011u8], &it.next().unwrap(), bits_pp);
+    splat_expand_pass(img, stride, &[0b11100100u8], &it.next().unwrap(), bits_pp);
+    splat_expand_pass(img, stride, &[0b00011011u8], &it.next().unwrap(), bits_pp);
+    splat_expand_pass(img, stride, &[0b11100100u8], &it.next().unwrap(), bits_pp);
+    copy_nbytes_mtimes(expected_img, &[0b10000001, 0b11100111], 2, 2);
+    copy_nbytes_mtimes(&mut expected_img[4..], &[0b00110110, 0b10011100], 2, 2);
+    copy_nbytes_mtimes(&mut expected_img[8..], &[0b01001001, 0b10101111], 2, 2);
+    copy_nbytes_mtimes(&mut expected_img[12..], &[0b11111010, 0b01010000], 2, 2);
+
+    assert_eq!(img, expected_img);
+
+    // Seventh pass
+    splat_expand_pass(
+        img,
+        stride,
+        &[0b00011011u8, 0b00011011u8],
+        &it.next().unwrap(),
+        bits_pp,
+    );
+    splat_expand_pass(
+        img,
+        stride,
+        &[0b11100100u8, 0b11100100u8],
+        &it.next().unwrap(),
+        bits_pp,
+    );
+    splat_expand_pass(
+        img,
+        stride,
+        &[0b00011011u8, 0b00011011u8],
+        &it.next().unwrap(),
+        bits_pp,
+    );
+    splat_expand_pass(
+        img,
+        stride,
+        &[0b11100100u8, 0b11100100u8],
+        &it.next().unwrap(),
+        bits_pp,
+    );
+    copy_nbytes_mtimes(&mut expected_img[2..], &[0b00011011u8, 0b00011011u8], 2, 1);
+    copy_nbytes_mtimes(&mut expected_img[6..], &[0b11100100u8, 0b11100100u8], 2, 1);
+    copy_nbytes_mtimes(&mut expected_img[10..], &[0b00011011u8, 0b00011011u8], 2, 1);
+    copy_nbytes_mtimes(&mut expected_img[14..], &[0b11100100u8, 0b11100100u8], 2, 1);
+
+    assert_eq!(img, expected_img);
+}
+
+#[test]
+fn test_splat_expand_pass_within_8x8() {
+    /*
+       [ 11, 12, 13, 14,
+         21, 22, 23, 24,
+         31, 32, 33, 34,
+         41, 42, 43, 44,
+         51, 52, 53, 54,
+         61, 62, 63, 64
+       ]
+    */
+
+    let actual_img = &mut vec![0u8; 24];
+    let expected_img = &mut vec![0u8; 24];
+    let img_row_stride = 4;
+    let bp_pixel = 8;
+
+    let mut it = Adam7Iterator::new(4, 6);
+
+    // First pass
+    expected_img[0..4].copy_from_slice(&vec![11, 11, 11, 11]);
+    repeat_nth_row_mtimes(expected_img, img_row_stride, 0, 5);
+
+    splat_expand_pass(
+        actual_img,
+        img_row_stride,
+        &[11],
+        &it.next().unwrap(),
+        bp_pixel,
+    );
+    assert_eq!(actual_img, expected_img);
+
+    // Third pass
```

Comment:
This confused me for a moment.  I wonder if you could please expand the comment to something like:

```
/// Third pass.  (Not "second pass" because for a 4-pixel-wide image `Adam7Iterator` will skip the no-op 2nd pass.)
```

Did I get that right in the proposed comment?